docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,566 @@
+  similar to the primary key in a relation database. By default the name of the key attribute for every object type
+  is set to "id". You can overwrite the name of the key attribute for a specific object type by calling
+  ``set_key_attribute_of_object()``.
+- ``get_representation_function_of_object``: Let's focus on the following restaurant:
+
+  .. code-block:: json
+
+      {
+          "id": 0,
+          "name": "Donath",
+          "cuisine": "Italian",
+          "outside-seating": true,
+          "price-range": "mid-range"
+      }
+
+  When the user is asking to list any Italian restaurant, you don't want to confront the user with all details of that
+  restaurant. You want to provide a meaningful name that identifies the restaurant. Most likely you would use
+  just the name of the restaurant to speak about it.
+  Thus, the function ``get_representation_function_of_object`` returns a lambda function that maps, for example, the
+  above restaurant object to its name.
+
+  .. code-block:: python
+
+      lambda obj: obj["name"]
+
+  This function is used whenever the bot is talking about a specific object, so that the user is given a meaningful
+  name and he knows what exactly the bot is talking about.
+  By default the lambda function is set to ``lambda obj: obj["name"]``. So, it returns the value of the attribute
+  "name" of the object. If your object does not have an attribute "name", or the "name" of an object might be
+  ambiguous, you should set a new lambda function for that object type by calling
+  ``set_representation_function_of_object()``.
+- ``set_ordinal_mention_mapping``: The ordinal mention mapping is needed to resolve an ordinal mention to an object
+  in a list. For example, if the bot listed a few restaurants in Berlin, and the user then asked
+
+    `Does the second one have outside seating?`
+
+  you need to resolve "second one" to the correct object the bot listed before. Per
+  default the ordinal mention mapping looks like this:
+
+  .. code-block:: python
+
+      {
+          "1": lambda l: l[0],
+          "2": lambda l: l[1],
+          "3": lambda l: l[2],
+          "4": lambda l: l[3],
+          "5": lambda l: l[4],
+          "6": lambda l: l[5],
+          "7": lambda l: l[6],
+          "8": lambda l: l[7],
+          "9": lambda l: l[8],
+          "10": lambda l: l[9],
+          "ANY": lambda l: random.choice(list),
+          "LAST": lambda l: l[-1],
+      }
+
+  You can overwrite it by calling the function ``set_ordinal_mention_mapping``.
+  If you want to learn more about the usage of the mapping, go to section :ref:`resolve_mentions`.
+
+
+
+Customize your `ActionQueryKnowledgeBase`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can overwrite the following two functions of `ActionQueryKnowledgeBase`:
+
+- ``utter_objects()``
+- ``utter_attribute_value()``
+
+``utter_objects()`` is used when the user requested the bot to list some objects.
```

Comment:
If I want to overwrite those functions, I would need to create a new knowledge base. I don't want that. I added doc strings to all methods and the `InMemoryKnowledgeBase` is basically an example implementation of the `KnowledgeBase` class. I think that is enough.
In the knowledgebasebot example I'm overwriting one of the functions. So, it should be clear how to do it.

Reply:
> If I want to overwrite those functions, I would need to create a new knowledge base. I don't want that. I added doc strings to all methods and the InMemoryKnowledgeBase is basically an example implementation of the KnowledgeBase class. I think that is enough.

Got it. Maybe we should link to the code where those methods are found

> In the knowledgebasebot example I'm overwriting one of the functions. So, it should be clear how to do it.

Similarly, reference that there is an example of this in the knowledgebasebot example since we're pretty far down on the page by now